CondTools/HLT/python/AlCaRecoTriggerBitsRcdUpdate_cfg_o2o.py

```diff
@@ -0,0 +1,182 @@
+                  VarParsing.VarParsing.varType.int,
+                  "the first run"
+                  )
+
+options.register( "hltKey", 
+                  "/cdaq/special/PilotBeamTest2021/Collisions/V55",  #default value
+                  VarParsing.VarParsing.multiplicity.singleton, 
+                  VarParsing.VarParsing.varType.string,
+                  "the hlt key"
+                  )
+
+options.register( "keyToModify", 
+                  "SiStripCalMinBias",  #default value
+                  VarParsing.VarParsing.multiplicity.singleton, 
+                  VarParsing.VarParsing.varType.string,
+                  "the key to modify"
+                  )
+
+options.register('pathsToModify',
+                 'HLT_ZeroBias_part*_v*', #default value
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.string,
+                 "Comma-separated list of paths to be modified")
+options.parseArguments()
+
+process.load("FWCore.MessageLogger.MessageLogger_cfi")
+process.MessageLogger.cerr = cms.untracked.PSet(enable = cms.untracked.bool(True))
+process.MessageLogger.cout = cms.untracked.PSet(INFO = cms.untracked.PSet(
+    reportEvery = cms.untracked.int32(1)
+    ))
+process.MessageLogger.cout.enable = cms.untracked.bool(True)
+process.MessageLogger.cout.threshold = cms.untracked.string('DEBUG')
+process.MessageLogger.debugModules = cms.untracked.vstring('*')
+# the module writing to DB
+from CondTools.HLT.alCaRecoTriggerBitsRcdUpdate_cfi import alCaRecoTriggerBitsRcdUpdate as _alCaRecoTriggerBitsRcdUpdate
+process.AlCaRecoTriggerBitsRcdUpdate = _alCaRecoTriggerBitsRcdUpdate.clone()
+# The IOV that you want to write out, defaut is 1 to -1/inf. 
+process.AlCaRecoTriggerBitsRcdUpdate.firstRunIOV = options.firstRun # docu see...
+#process.AlCaRecoTriggerBitsRcdUpdate.lastRunIOV = -1 # ...cfi
+# If you want to start from scratch, comment the next line:
+process.AlCaRecoTriggerBitsRcdUpdate.startEmpty = False
+
+start = time.time()
+
+print('HLT menu:', options.hltKey)
+
+cmd = "hltGetConfiguration adg:%s" %options.hltKey
+p = subprocess.Popen( cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT )
+out, err = p.communicate()
+
+# if hltGetConfiguration fails
+if p.returncode != 0:
+    raise Exception(out.decode('utf-8'))
+
+# get part where pds:paths are defined
+is_block = False
+lines=''
+for line in out.decode('utf-8').split('\n'):
+  if is_block and 'process.' in line:
+    break  
+  if 'process.datasets' in line:
+    is_block = True
+  if is_block:
+    lines += line+"\n"
+
+exec(lines)
+
+pds = process.datasets._Parameterizable__parameterNames
+for pd in pds:
+  paths = sorted( path for path in process.datasets.__dict__[pd] )
+  if (re.search("^ZeroBias\d{1,2}", pd)): # only search for 'ZeroBiasX'
+    pattern = 'HLT_*ZeroBias*part*'
+    matching = fnmatch.filter(paths, pattern)
+    if matching:
+      print(pd, ':', *matching)
```

Comment:
Maybe it would be slightly better to check the streams as follows
```py
re.match('PhysicsZeroBias([0-9]+)$', foo)
```
(one could in principle have a config with datasets defined, but not assigned to any streams, although the latest ConfDB probably has a check against this).

The regex in the current impl could nevertheless be improved, i think (note that for "split" datasets ConfDB now makes the name end with numbers, thus the `$`).

Maybe most important: the HLT handling of datasets is slightly changing in Run 3, and (long story short) the menu might not have anymore paths called `HLT_ZeroBias_part*_v*`; one would instead find only `HLT_ZeroBias_v`, plus so-called Dataset Paths named `Dataset_ZeroBias([0-9]+)`. I don't know when and how this is propagated to online, but I suggest to coordinate with FOG in advance.

Reply:
> Maybe most important: the HLT handling of datasets is slightly changing in Run 3, and (long story short) the menu might not have anymore paths called `HLT_ZeroBias_part*_v*`; one would instead find only `HLT_ZeroBias_v`, plus so-called Dataset Paths named `Dataset_ZeroBias([0-9]+)`. I don't know when and how this is propagated to online, but I suggest to coordinate with FOG in advance.

If that's the case, some of the reasons to introduce the O2O fall down, because there won't be necessity to roll back and forth from `HLT_ZeroBias_*`  to `HLT_ZeroBias_v*`. The `AlCaRecoTriggerBit` could just be configured to take the union of the two sets.
Where can one learn more about it?

Reply:
The most recent presentation I remember is [this](https://indico.cern.ch/event/1111793/contributions/4675649/attachments/2370202/4048016/confdb_output_changes.pdf). Silvio might have better refs (or one might have to ask Sam).

Reply:
The documentation on the new dataset workflow (from @Sam-Harper) is here: https://twiki.cern.ch/twiki/bin/viewauth/CMS/ConfDBRun3#PathDefined_Datasets_and_FinalPa

It has been implemented in our online menus: [CMSHLT-2234](https://its.cern.ch/jira/browse/CMSHLT-2234) (but not yet used in the context of split datasets)